lib/src/main/java/org/altbeacon/beacon/service/ScanJob.java

```diff
@@ -136,15 +136,12 @@ private void startPassiveScanIfNeeded() {
         }
         if (insideAnyRegion) {
             // TODO: Set up a scan filter for not detecting a beacon pattern
-            LogManager.i(TAG, "We are inside a beacon region.  We will not scan between cycles.");
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            mScanHelper.startAndroidOBackgroundScan(mScanState.getBeaconParsers());
```

Comment:
I don't think this code change will be helpful.  If you are already inside a beacon region (meaning the beacon is visible in the vicinity), there is no point in calling `mScanHelper.startAndroidOBackgroundScan(mScanState.getBeaconParsers());`.  This will just result in callbacks quickly each time that an already visible beacon continues to be seen.  This will accomplish nothing more than using extra CPU and battery when the app is in the background.